tests/test_db_constraints.py

```diff
@@ -0,0 +1,569 @@
+# IMPORTS ################################################################################ IMPORTS #
+
+# Standard library
+
+# Installed
+from typing import no_type_check
+import pytest
+import sqlalchemy
+
+# Own
+import tests
+import dds_web
+from dds_web import db
+from dds_web.database import models
+
+# TESTS #################################################################################### TESTS #
+
+
+# Unit ###################################################################################### Unit #
+
+
+def __setup_unit():
+    unit = models.Unit.query.filter_by(name="Unit 1").first()
+
+    projects = unit.projects
+    assert projects != []
+
+    unit_users = unit.users
+    assert unit_users != []
+
+    return unit, projects, unit_users
+
+
+def __delete_project(project):
+    for version in project.file_versions:
+        db.session.delete(version)
+
+    for file in project.files:
+        db.session.delete(file)
+
+    db.session.delete(project)
+
+    db.session.commit()
+
+
+def test_delete_unit_row__with_project_and_users(client):
```

Comment:
Already mentioned but adding here too: 
Make sure to test with empty projects, currently the projects contain files.